examples/07_tutorials/ai-reco-of-recos/utils/file_utils.py

```diff
@@ -0,0 +1,309 @@
+import requests
+import urllib
+import os
+import pickle
+import logging
+import base64
+from PIL import Image
+from datetime import datetime, timedelta
+
+
+def locate_prompt(prompt_name, module_directory):
+    """
+    Locates a prompt file in the prompts directory.
+
+    Args:
+        prompt_name (str): The name of the prompt file.
+
+    Returns:
+        str: The full path to the prompt file.
+    """
+    prompt_path = os.path.join(module_directory, 'prompts', prompt_name)
+    if os.path.exists(prompt_path): return prompt_path
+
+    prompt_path = os.path.join('prompts', prompt_name) 
+    if os.path.exists(prompt_path): return prompt_path
+
+    prompts_directory = locate_directory('prompts')
+    if prompts_directory:
+        prompt_path = os.path.join(prompts_directory, prompt_name)
+        if os.path.exists(prompt_path): return prompt_path
+        
+    return os.path.join(module_directory, 'prompts', prompt_name)
```

Comment:
will this filepath always exist? If not, raising ValueError or FileNotFoundError seems the right choice.